### DISCUSSION TITLE
Short syntax for named tuple elements

### DISCUSSION THREAD

**Discussion by tuscen** [dotnet/csharplang#6096] [Language Ideas] [answered]
Often tuples with named elements are much easier to read, but the element's and variable's names often match, so it's inconvenient to write such tuples:

```csharp
var foo = 1;
var baz = 2;
var quux = 3;
var tuple = (foo: foo, bar: bar, quux: quux);
```

It would be more convenient to have a shorter syntax for this, for example:

```csharp
var tuple = (foo:, bar:, quux:);
```

JS/TS have something similar, but for objects:

```js
let foo = 1;
let bar = 3;
let quux = 3;

let o1 = { foo: foo, bar: bar, quux: quux };
let o2 = { foo, bar, quux };
```

`o1` and `o2` are equivalent.

But since `(foo, bar, quux)` is already a valid syntax for creating a tuple from these variables, a colon could be added after the variable name to tell the compiler to use the variable's name and it's value at the same time to compose a named tuple element.

**Comment by jnm2** [comment, marked as answer]
Why is `(foo: foo, bar: bar, quux: quux)` or `(foo:, bar:, quux:)` preferable over the existing and shorter `(foo, bar, quux)`?

```cs
var foo = 1;
var bar = 2;
var quux = 3;
var tuple = (foo, bar, quux);
// This already gives tuple element names:
_ = tuple.foo;
_ = tuple.bar;
_ = tuple.quux;
```

**Reply by tuscen** [reply]
You have to use generic `ItemN` properties to refer to tuples elements. That way you loose the semantic what is the elements purpose

**Reply by jnm2** [reply]
@tuscen If you try it, you can see that this is not correct. See the example I added in.

**Reply by HaloFour** [reply]
[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7NAExAGoAfAAQCYBGAWAChGKBmAAmrYGE2BvRtoI5UqfAUIkBLPBjYI2AXjY0A3OImDpsgJ6K2VNQw1CAbgEMobDDgAOAGxh6AFAjRttASkPHB6jRRoATicAEgAiJz5rexgAOgQAXzdeaIdY7QS2DzCvPwTGBKA)

**Reply by tuscen** [reply]
Sorry, I've got confused somehow

**Reply by HaloFour** [reply]
Tuples in C# 7.0 didn't originally infer names like this, it was added in C# 7.1.

**Comment by HaloFour** [comment]
Relevant: https://github.com/dotnet/csharplang/blob/main/proposals/csharp-7.1/infer-tuple-names.md
